src/main/redux/sagas/reader.ts

```diff
@@ -193,58 +198,80 @@ export function* readerCloseRequestWatcher(): SagaIterator {
         const action = yield take(readerActions.ActionType.CloseRequest);
         const reader = action.payload.reader;
         const gotoLibrary = action.payload.gotoLibrary;
-        const publication = reader.publication;
 
-        // Notify the streamer that a publication has been closed
-        yield put(streamerActions.closePublication(
-            publication,
-        ));
+        const closeFunction = closeReader.bind(closeReader, reader, gotoLibrary);
+        yield call(closeFunction);
+    }
+}
 
-        // Wait for the publication to be closed
-        const streamerAction = yield take([
-            streamerActions.ActionType.PublicationCloseSuccess,
-            streamerActions.ActionType.PublicationCloseError,
-        ]);
+export function* closeReaderFromPublicationWatcher(): SagaIterator {
+    while (true) {
+        const action = yield take(readerActions.ActionType.CloseFromPublicationRequest);
+        const publication = action.payload.publication;
+        const store: any = container.get("store");
+        const readers = (store.getState().reader as ReaderState).readers;
 
-        if (streamerAction.error) {
-            // Failed to close publication
-            yield put({
-                type: readerActions.ActionType.CloseError,
-                payload: {
-                    reader,
-                },
-            });
-            continue;
+        for (const reader of Object.values(readers)) {
+            if (reader.publication.identifier === publication.identifier) {
+                const closeFunction = closeReader.bind(closeReader, reader, false);
+                yield call(closeFunction);
+            }
         }
+    }
+}
 
-        const winRegistry = container.get("win-registry") as WinRegistry;
-        const readerWindow = winRegistry.getWindowByIdentifier(reader.identifier);
-
-        if (gotoLibrary) {
-            // Show library window
-            const appWindows = winRegistry.getWindows();
-
-            for (const appWin of Object.values(appWindows)) {
-                if (appWin.type !== AppWindowType.Library) {
-                    continue;
-                }
+function* closeReader(reader: any, gotoLibrary: boolean): any {
```

Comment:
Can this `any` be replaced with a proper TypeScript type?
`Reader` from `src/common/models/reader.ts`?
https://github.com/readium/readium-desktop/blob/7776294d83aae7184eee47ccfa06adde13789d14/src/common/models/reader.ts#L21-L26

Reply:
I deleted the return type, it was foor something I wanted to do but I choosed an other way to do it and didn't delete it

Reply:
Okay thank. What about `reader: any`?

Reply:
Oh, sorry, I didn't saw it. I fixed it